packages/api/src/platforms/vtex/loaders/pagetype.ts

```diff
@@ -0,0 +1,26 @@
+import DataLoader from 'dataloader'
+import pLimit from 'p-limit'
+
+import type { Options } from '..'
+import type { Clients } from '../clients'
+import type { PortalPagetype } from '../clients/commerce/types/Portal'
+
+// Limits concurrent requests to 20 so that they don't timeout
+const CONCURRENT_REQUESTS_MAX = 20
+
+export const getPagetypeLoader = (_: Options, clients: Clients) => {
+  const limit = pLimit(CONCURRENT_REQUESTS_MAX)
+
+  const loader = (slugs: readonly string[]) => {
+    return Promise.all(
+      slugs.map((s: string) =>
```

Comment:
why not using p-map in here? this way we don't need this Promise.all chained with a map chained with a limit

Reply:
p-limit-ing there would allow more than 20 requests to be executed at the same time, and we want to avoid just that.